File: src/sdk/go/hanami_sdk/task.go
/**
 * @author      Tobias Anker <tobias.anker@kitsunemimi.moe>
 *
 * @copyright   Apache License Version 2.0
 *
 *      Copyright 2022 Tobias Anker
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *          http://www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 */

package hanami_sdk

type TaskInput struct {
	HexagonName        string `json:"hexagon_name"`
	DatasetColumnName  string `json:"dataset_column"`
	DatasetUuid        string `json:"dataset_uuid"`
}

type TaskResult struct {
	HexagonName        string `json:"hexagon_name"`
	DatasetColumnName  string `json:"dataset_column"`
}

func CreateTrainTask(address, token, name, clusterUuid string, inputs, outputs []TaskInput, timeLenght int, skipTlsVerification bool) (map[string]interface{}, error) {
    var inputArray []interface{}
    for _, input := range inputs {
        inputArray = append(inputArray, input)
    }

    var outputArray []interface{}
    for _, output := range outputs {
        outputArray = append(outputArray, output)
    }

	path := "v1.0alpha/task/train"
	jsonBody := map[string]interface{}{
		"name":         name,
		"cluster_uuid": clusterUuid,
		"inputs":       inputArray,
		"outputs":      outputArray,
		"time_length":  timeLenght,
	}
	return SendPost(address, token, path, jsonBody, skipTlsVerification)
}

func CreateRequestTask(address, token, name, clusterUuid string, inputs []TaskInput, results []TaskResult, timeLenght int, skipTlsVerification bool) (map[string]interface{}, error) {
	var inputArray []interface{}
    for _, input := range inputs {
        inputArray = append(inputArray, input)
    }

    var resultArray []interface{}
    for _, result := range results {
        resultArray = append(resultArray, result)
    }

	path := "v1.0alpha/task/request"
	jsonBody := map[string]interface{}{
		"name":         name,
		"cluster_uuid": clusterUuid,
		"inputs":       inputArray,
		"results":      resultArray,
		"time_length":  timeLenght,
	}
	return SendPost(address, token, path, jsonBody, skipTlsVerification)
}

func GetTask(address, token, taskId, clusterUuid string, skipTlsVerification bool) (map[string]interface{}, error) {
	path := "v1.0alpha/task"
	vars := map[string]interface{}{
		"uuid":         taskId,
		"cluster_uuid": clusterUuid,
	}
	return SendGet(address, token, path, vars, skipTlsVerification)
}

func ListTask(address, token, clusterUuid string, skipTlsVerification bool) (map[string]interface{}, error) {
	path := "v1.0alpha/task/all"
	vars := map[string]interface{}{"cluster_uuid": clusterUuid}
	return SendGet(address, token, path, vars, skipTlsVerification)
}

func DeleteTask(address, token, taskUuid, clusterUuid string, skipTlsVerification bool) (map[string]interface{}, error) {
	path := "v1.0alpha/task"
	vars := map[string]interface{}{
		"uuid":         taskUuid,
		"cluster_uuid": clusterUuid,
	}
	return SendDelete(address, token, path, vars, skipTlsVerification)
}

File: src/sdk/python/hanami_sdk/task.py
# Copyright 2022 Tobias Anker
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from hanami_sdk import hanami_request
import json


def create_task(token: str,
                address: str, 
                name: str,
                task_type: str, 
                cluster_uuid: str,
                dataset_uuid: str) -> tuple[bool,str]:
    path = "/control/v1/task"
    json_body = {
        "name": name,
        "type": task_type,
        "cluster_uuid": cluster_uuid,
        "dataset_uuid": dataset_uuid,
    }
    body_str = json.dumps(json_body)
    return hanami_request.send_post_request(token, address, path, body_str)


def get_task(token: str, address: str, task_uuid: str, cluster_uuid: str) -> tuple[bool,str]:
    path = "/control/v1/task"
    values = f'uuid={task_uuid}&cluster_uuid={cluster_uuid}'
    return hanami_request.send_get_request(token, address, path, values)


def list_tasks(token: str, address: str, cluster_uuid: str) -> tuple[bool,str]:
    path = "/control/v1/task/all"
    values = f'cluster_uuid={cluster_uuid}'
    return hanami_request.send_get_request(token, address, path, values)


def delete_task(token: str, address: str, task_uuid: str, cluster_uuid: str) -> tuple[bool,str]:
    path = "/control/v1/task"
    values = f'uuid={task_uuid}&cluster_uuid={cluster_uuid}'
    return hanami_request.send_delete_request(token, address, path, values)
